src/permission/fs_permission.cc

```diff
@@ -211,7 +211,7 @@ void FSPermission::RadixTree::Insert(const std::string& path) {
 
     if (is_wildcard_node || is_last_char) {
       std::string node_path = path.substr(parent_node_prefix_len, i);
-      current_node = current_node->CreateChild(node_path);
+      current_node = current_node->CreateChild(std::move(node_path));
```

Comment:
This doesn't really change anything (except maybe shut up coverity) because CreateChild takes a value, not a reference. What you want instead is this:
```diff
diff --git a/src/permission/fs_permission.h b/src/permission/fs_permission.h
index 244e95727a..80374a08c7 100644
--- a/src/permission/fs_permission.h
+++ b/src/permission/fs_permission.h
@@ -31,7 +31,7 @@ class FSPermission final : public PermissionBase {
 
       Node() : wildcard_child(nullptr), is_leaf(false) {}
 
-      Node* CreateChild(std::string prefix) {
+      Node* CreateChild(const std::string& prefix) {
         if (prefix.empty() && !is_leaf) {
           is_leaf = true;
           return this;
```
Or `Node* CreateChild(std::string&& prefix) {` + `std::move(...)`. Look at other call sites to determine which is most appropriate.